imageio/imageio-dds/src/main/java/com/twelvemonkeys/imageio/plugins/dds/DDSHeader.java

```diff
@@ -0,0 +1,140 @@
+package com.twelvemonkeys.imageio.plugins.dds;
+
+import javax.imageio.IIOException;
+import javax.imageio.stream.ImageInputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.ByteOrder;
+import java.util.Arrays;
+
+final class DDSHeader {
+
+    // https://learn.microsoft.com/en-us/windows/win32/direct3ddds/dx-graphics-dds-pguide
+    private int flags;
+    private int width;
+    private int height;
+    private int mipmap;
+
+    private int pixelFormatFlags;
+    private int fourCC;
+    private int bitCount;
+    private int redMask;
+    private int greenMask;
+    private int blueMask;
+    private int alphaMask;
+
+    public static DDSHeader read(final ImageInputStream imageInput) throws IOException {
+        DDSHeader header = new DDSHeader();
+        
+        // Read MAGIC bytes [0,3]
+        byte[] magic = new byte[DDS.MAGIC.length];
+        imageInput.readFully(magic);
+        if (!Arrays.equals(DDS.MAGIC, magic)) {
+            throw new IIOException(String.format("Not a DDS file. Expected DDS magic %s, read %s", String.format("%08x", new BigInteger(1, DDS.MAGIC)), String.format("%08x", new BigInteger(1, magic))));
```

Comment:
Sorry, I forgot that the magic was a byte array, not an int... 

I think, however, this should work:

```suggestion
            throw new IIOException(String.format("Not a DDS file. Expected DDS magic %08x, read %08x", new BigInteger(1, DDS.MAGIC), new BigInteger(1, magic)));
```

Reply:
I did something similar...
`String.format("%08x", new BigInteger(1, magic))`

Reply:
Yes, but that is 3 `String.format`s instead of 1...